### PR TITLE
[WIP] fix(`revm`): several fixes on commits

### PR DESCRIPTION
**Pull request by Evalir**

### CODE REVIEW

**Comment by Evalir** [comment]
Closing—these bugs need to be chased in revm first

File: crates/evm/src/executor/fork/cache.rs [hunk 1]
```diff
@@ -267,22 +273,13 @@ impl MemDb {
...
-                    if value.present_value() == U256::from(0) {
-                        acc_storage.remove(&index);
-                    } else {
-                        acc_storage.insert(index, value.present_value());
-                    }
-                }
-                if acc_storage.is_empty() {
-                    storage.remove(&add);
-                }
+                acc_storage.extend(acc.storage.into_iter().map(|(k, v)| (k, v.present_value())));
```
Comment (rakita) [inline crates/evm/src/executor/fork/cache.rs:282]: I would leave the previous code here, as it does some cleanup of current storage. It probably didn't change any behaviour but it uses a little bit less memory.

File: crates/evm/src/executor/fork/cache.rs [hunk 2]
```diff
@@ -251,10 +251,16 @@ impl MemDb {
...
-                accounts.remove(&add);
-                storage.remove(&add);
+            if !acc.is_touched() {
+                continue
+            }
+            if acc.is_selfdestructed() {
+                let acc_storage = storage.entry(add).or_default();
+                acc_storage.clear();
+                let acc_info = Default::default();
+                accounts.insert(add, acc_info);
```
Comment (rakita) [inline crates/evm/src/executor/fork/cache.rs:261]: Would leave old behaviour here as it is cleaning some data.

File: crates/evm/src/executor/fork/cache.rs [hunk 3]
```diff
@@ -251,10 +251,16 @@ impl MemDb {
         let mut storage = self.storage.write();
         let mut accounts = self.accounts.write();
         for (add, mut acc) in changes {
-            if acc.is_empty() || acc.is_selfdestructed() {
-                accounts.remove(&add);
-                storage.remove(&add);
+            if !acc.is_touched() {
```
Comment (rakita) [inline crates/evm/src/executor/fork/cache.rs:254]: This should stay, it is a small improvement as filtering is not done inside revm and all loaded data is returned from revm.

**Pull request closed without merging**